src/coffee/directives/api/models/parent/circle-parent-mode.coffee

```diff
@@ -21,13 +28,35 @@ angular.module('uiGmapgoogle-maps.directives.api.models.parent')
       ]
       @watchProps()
 
-      listeners = @setEvents gObject, scope, scope
+      clean()
+      @listeners = @setEvents gObject, scope, scope, ['radius_changed']
 
-      google.maps.event.addListener gObject, 'radius_changed', ->
-        scope.$evalAsync ->
-          scope.radius = gObject.getRadius()
+      @listeners.push google.maps.event.addListener gObject, 'radius_changed', ->
+        ###
+          possible google bug, and or because a circle has two radii
+          radius_changed appears to fire twice (original and new) which is not too helpful
+          therefore we will check for radius changes manually and bail out if nothing has changed
+        ###
+
+        newRadius = gObject.getRadius()
+        return if newRadius == lastRadius
+        console.debug "newRadius: #{newRadius}, oldRadius: #{lastRadius}"
+        lastRadius =  newRadius
+
+        work = ->
+          console.error 'radius_changed evalAsync'
+          scope.radius = newRadius if newRadius != scope.radius
+          scope.events.radius_changed(gObject, 'radius_changed', scope, arguments) if scope.events?.radius_changed and _.isFunction scope.events?.radius_changed
+
+        # hack
+        # for some reason in specs I can not get $evalAsync to fire.. im tired of wasting time on this
```

Comment:
We can flush this out more later. I want to move on from this PR. Currently it works.. even though it is not the cleanest.